bson/src/main/org/bson/codecs/pojo/PojoBuilderHelper.java

```diff
@@ -42,11 +43,18 @@
 import static org.bson.codecs.pojo.PropertyReflectionUtils.toPropertyName;
 
 final class PojoBuilderHelper {
-
-    @SuppressWarnings("unchecked")
     static <T> void configureClassModelBuilder(final ClassModelBuilder<T> classModelBuilder, final Class<T> clazz) {
         classModelBuilder.type(notNull("clazz", clazz));
 
+        if (isRecord(clazz)) {
```

Comment:
@jyemin in the design doc you suggested "inverting the dependency" so that whoever constructed the ClassModelBuilder was responsible for configuring it correctly (i.e. RecordCodecProvider calls configureRecordClassModelBuilder). I implemented this initially, but I realized this would introduce a breaking change. 

ClassModel::builder is part of the public API and it constructs a ClassModelBuilder which originally called the configureClassModelBuilder method. To invert the dependency we would get rid of this call. Previously users get a fully configured classModelBuilder just by calling ClassModel#builder, now it isn't configured it all. This is bad because we accept ClassModels in our PojoCodecProvider in one of the register functions. I believe this means we can't invert the dependency so I did it this way instead.